lib/kafka/consumer.rb

```diff
@@ -73,6 +73,10 @@ def initialize(cluster:, logger:, instrumenter:, group:, fetcher:, offset_manage
       #   when user commits message other than last in a batch, this would make ruby-kafka refetch
       #   some already consumed messages
       @current_offsets = Hash.new { |h, k| h[k] = {} }
+
+      # Hash storing topics that are already being subscribed
+      # When subcribing to a new topic, if it's already being subscribed before, skip it
+      @subscribed_topics = {}
```

Comment:
Can you use `Set` here instead?

Reply:
@dasch I updated my PR as your recommendation.